lib/withLoggedInUser.js

```diff
@@ -50,7 +50,10 @@ const withLoggedInUser = WrappedComponent => {
     };
 
     getLoggedInUserFromServer = () => {
-      return this.props.client.query({ query: loggedInUserQuery, fetchPolicy: 'network-only' }).then(result => {
+      // Notice that fetchPolic is set to 'no-cache'. This is similar to network-only, except the query's result is not stored in the cache.
+      // This is an important detail when fetching for inherited memberShip of projects and events which don't necessarily have a MemberId.
```

Comment:
The fact that they don't have a `MemberId` is an API problem that we should resolve. Not storing the `LoggedInUser` in the cache can have undesired side effects, and having null IDs can later cause issues. This workaround is a blocker for me.

Reply:
I don't think it is such a huge performance issue due to the fact the previous fetch policy would still require the query to fetch new data before resolving.
But I agree that this is an ugly workaround the real issue: https://github.com/opencollective/opencollective-api/pull/7196#discussion_r813878174.

Reply:
This call has `network-only` because it's about refetching loggedinuser from the server. But https://github.com/opencollective/opencollective-frontend/blob/main/components/changelog/ChangelogTrigger.js#L43 for example relies on having the logged in user stored in cache, and with this update it would required the trigger of a second query to work. Actually, I'm curious if it would work at all since it has a `cachePolicy: cache-only`.